crates/cast/bin/cmd/send.rs

```diff
@@ -197,20 +197,23 @@ impl SendTxArgs {
             // move the signers to their respective blocks.
             if zksync_params.zksync {
                 // Retrieve the signer, and bail if it can't be constructed.
-                let zk_signer: foundry_wallets::WalletSigner = eth.wallet.signer().await?;
-                let from = zk_signer.address();
+                let signer = eth.wallet.signer().await?;
+                let from = signer.address();
 
                 tx::validate_from_address(eth.wallet.from, from)?;
 
                 // Zksync transaction
-                let (tx, _) = builder.build(&zk_signer).await?;
+                let (tx, _) = builder.build(&signer).await?;
+                let signer = Arc::new(signer);
 
-                let zk_wallet = ZksyncWallet::from(zk_signer);
+                let zk_wallet = ZksyncWallet::from(signer.clone());
```

Comment:
Didn't you say this panics in ledger?

Reply:
The initialization is, not sure about a clone, but I put it behind an Arc to ensure we don’t actually clone the data